File: loan-repayment-java/src/main/java/com/loan/repayment/entity/RepaymentPlanRequest.java
package com.loan.repayment.entity;

import java.time.ZonedDateTime;

import javax.validation.constraints.NotNull;

import org.springframework.validation.annotation.Validated;

import io.swagger.annotations.ApiModelProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * 
 * @author Ramachandra
 *
 */
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Getter
@EqualsAndHashCode
@Validated
public class RepaymentPlanRequest {

    /**
     * The loan amount.
     */
    @NotNull
    @ApiModelProperty(notes = "The loan amount", example = "5000", required = true)
    private String loanAmount;

    /**
     * The nominal rate of interest for year.
     */
    @NotNull
    @ApiModelProperty(notes = "The nominal rate of interest " + "for year", example = "5", required = true)
    private String nominalRate;

    /**
     * The loan repayment duration.
     */
    @NotNull
    @ApiModelProperty(notes = "The loan repayment" + " duration", example = "24", required = true)
    private int duration;

    /**
     * The loan payment start date.
     */
    @NotNull
    @ApiModelProperty(notes = "The loan payment start date", required = true)
    private ZonedDateTime startDate;

}

File: loan-repayment-kotlin/src/main/kotlin/com/loan/repayment/entity/RepaymentPlanRequest.kt
package com.loan.repayment.entity

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.annotations.ApiModelProperty
import java.time.ZonedDateTime

/**
 *
 * @author Rama chandra
 */
data class RepaymentPlanRequest(
        @ApiModelProperty(notes = "The loan amount" , example = "5000" , required = true)
        @JsonProperty("loanAmount")
        var loanAmount : String ,
        @ApiModelProperty(notes = "The nominal rate of interest " + "for year" , example = "5" , required = true)
        @JsonProperty("nominalRate")
        var nominalRate : String ,
        @ApiModelProperty(notes = "The loan repayment" + " duration" , example = "24" , required = true)
        @JsonProperty("duration")
        var duration : Int ,
        @ApiModelProperty(notes = "The loan payment start date" , required = true)
        @JsonProperty("startDate")
        var startDate : ZonedDateTime
)